smaug/core/network_builder.cpp

```diff
@@ -117,10 +118,12 @@ static void createAndAddOperator(const NodeProto& node,
     dout(0) << "Adding " << name << " (" << OpType_Name(type) << ").\n";
 
     if (type == OpType::Data) {
-        const TensorData& tensorData = searchTensorDataArray(
+        const TensorData* tensorData = searchTensorDataArray(
                 tensorDataArray, node.input_tensors(0).name());
+        assert(tensorData &&
```

Comment:
This is a failure due to user input, so assert is not the most appropriate failure mechanism - this error message doesn't tell the user anything about how to fix the issue. It would be better to return error codes or statuses, but that would require a lot of plumbing. You can do a std::cerr print message and then quit.

Reply:
Thanks for the clarification! I was never clear about when to use assert or cerr.